stage0/src/usart.rs

```diff
@@ -0,0 +1,96 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use lpc55_pac as device;
+
+pub fn setup(
+    syscon: &device::syscon::RegisterBlock,
+    iocon: &device::iocon::RegisterBlock,
+    flexcomm: &device::flexcomm0::RegisterBlock,
+) {
+    gpio_setup(syscon, iocon);
+    flexcomm0_setup(syscon, flexcomm);
+}
+
+pub fn teardown(
+    syscon: &device::syscon::RegisterBlock,
+    iocon: &device::iocon::RegisterBlock,
+    flexcomm: &device::flexcomm0::RegisterBlock,
+) {
+    gpio_teardown(syscon, iocon);
+    flexcomm0_teardown(syscon, flexcomm);
+}
+/// Configure GPIO pin 29 & 30 for RX & TX respectively, as well as
+/// digital mode.
+fn gpio_setup(
+    syscon: &device::syscon::RegisterBlock,
+    iocon: &device::iocon::RegisterBlock,
+) {
+    // IOCON: enable clock & reset
+    syscon.ahbclkctrl0.modify(|_, w| w.iocon().enable());
+    syscon.presetctrl0.modify(|_, w| w.iocon_rst().released());
+
+    // GPIO: enable clock & reset
+    syscon.ahbclkctrl0.modify(|_, w| w.gpio0().enable());
+    syscon.presetctrl0.modify(|_, w| w.gpio0_rst().released());
+
+    // configure GPIO pin 29 & 30 for RX & TX respectively, as well as
+    // digital mode
+    iocon
+        .pio0_29
+        .write(|w| w.func().alt1().digimode().digital());
+    iocon
+        .pio0_30
+        .write(|w| w.func().alt1().digimode().digital());
+
+    // disable IOCON clock
+    syscon.ahbclkctrl0.modify(|_, w| w.iocon().disable());
+}
+
+fn gpio_teardown(
+    syscon: &device::syscon::RegisterBlock,
+    iocon: &device::iocon::RegisterBlock,
+) {
+    // IOCON: enable clock & reset
+    syscon.ahbclkctrl0.modify(|_, w| w.iocon().enable());
+    syscon.presetctrl0.modify(|_, w| w.iocon_rst().released());
+
+    // reset pin 29 and 30 to defaults
+    iocon.pio0_29.reset();
+    iocon.pio0_30.reset();
+
+    syscon.presetctrl0.modify(|_, w| w.iocon_rst().asserted());
+    syscon.ahbclkctrl0.modify(|_, w| w.iocon().disable());
+
+    syscon.presetctrl0.modify(|_, w| w.gpio0_rst().asserted());
+    syscon.ahbclkctrl0.modify(|_, w| w.gpio0().disable());
```

Comment:
again, probably not worth turning those off

Reply:
I figured the safest thing to do was to return the platform to the default state. This was intended to avoid possibly breaking assumptions made by existing hubris code but also to prevent future code in hubris relying on behavior in stage0 that may change. This may be overly cautious but unless we need to reduce code size or speed up boot time it seemed like the prudent thing to do.

Reply:
LMK if you're asking for this to be removed before merging.

Reply:
fixed